plugins/outputs/dynatrace/dynatrace.go

```diff
@@ -101,163 +88,92 @@ func (d *Dynatrace) Description() string {
+			switch tm.Type() {
+			case telegraf.Histogram:
+				if tag.Key == "le" || tag.Key == "gt" {
 					continue
 				}
-				if len(metric.Tags()[k]) > 0 {
-					fmt.Fprintf(&tagb, ",%s=%s", strings.ToLower(tagKey), d.escape(metric.Tags()[k]))
+			case telegraf.Summary:
+				if tag.Key == "quantile" {
+					continue
 				}
 			}
+			dims = append(dims, dimensions.NewDimension(string(tag.Key), tag.Value))
 		}
-		if len(metric.Fields()) > 0 {
-			for k, v := range metric.Fields() {
-				var value string
-				switch v := v.(type) {
-				case string:
-					continue
-				case float64:
-					if !math.IsNaN(v) && !math.IsInf(v, 0) {
-						value = fmt.Sprintf("%f", v)
-					} else {
-						continue
-					}
-				case uint64:
-					value = strconv.FormatUint(v, 10)
-				case int64:
-					value = strconv.FormatInt(v, 10)
-				case bool:
-					if v {
-						value = "1"
-					} else {
-						value = "0"
-					}
-				default:
-					d.Log.Debugf("Dynatrace type not supported! %s", v)
-					continue
-				}
 
-				// metric name
-				metricKey, err := d.normalize(k, maxMetricKeyLen)
-				if err != nil {
-					continue
+		metricType := tm.Type()
+		for _, field := range tm.FieldList() {
+			for _, i := range d.AddCounterMetrics {
+				if tm.Name()+"."+field.Key == i {
+					metricType = telegraf.Counter
 				}
+			}
 
-				metricID, err := d.normalize(d.Prefix+metric.Name()+"."+metricKey, maxMetricKeyLen)
-				// write metric name combined with its field
-				if err != nil {
-					continue
-				}
-				// write metric id,tags and value
+			typeOpt := getTypeOption(metricType, field)
 
-				metricType := metric.Type()
-				for _, i := range d.AddCounterMetrics {
-					if metric.Name()+"."+metricKey == i {
-						metricType = telegraf.Counter
-					}
-				}
+			if typeOpt == nil {
+				// unsupported type
+				continue
+			}
 
-				switch metricType {
-				case telegraf.Counter:
-					var delta float64
-
-					// Check if LastValue exists
-					if lastvalue, ok := d.State[metricID+tagb.String()]; ok {
-						// Convert Strings to Floats
-						floatLastValue, err := strconv.ParseFloat(lastvalue, 32)
-						if err != nil {
-							d.Log.Debugf("Could not parse last value: %s", lastvalue)
-						}
-						floatCurrentValue, err := strconv.ParseFloat(value, 32)
-						if err != nil {
-							d.Log.Debugf("Could not parse current value: %s", value)
-						}
-						if floatCurrentValue >= floatLastValue {
-							delta = floatCurrentValue - floatLastValue
-							fmt.Fprintf(&buf, "%s%s count,delta=%f\n", metricID, tagb.String(), delta)
-						}
-					}
-					d.State[metricID+tagb.String()] = value
-
-				default:
-					fmt.Fprintf(&buf, "%s%s %v\n", metricID, tagb.String(), value)
-				}
+			name := tm.Name() + "." + field.Key
+			dm, err := dtMetric.NewMetric(
+				name,
+				dtMetric.WithPrefix(d.Prefix),
+				dtMetric.WithDimensions(
+					dimensions.MergeLists(
+						// dimensions.NewNormalizedDimensionList(e.opts.DefaultDimensions...),
```

Comment:
```suggestion
```

Reply:
I left this there as a reminder. We currently don't have the default dimensions functionality. I was going to add it in a separate PR

Reply:
Oh, I see. Feel free to ignore this then!